### PR TITLE
PICMI: 0.16 & 0.18 & WarpX 1D

### PR DESCRIPTION
**Pull request by ax3l**
Update the `py-picmistandard` and the latest WarpX release.
Preparing 1D support (testable inputs coming for 22.01+).

Follow-up to #27800

### CODE REVIEW

**Review by dpgrote — DISMISSED** [review]
The changes look good to me.

File: var/spack/repos/builtin/packages/warpx/package.py [hunk 1]
```diff
@@ -164,7 +168,7 @@ def _get_input_options(self, post_install):
             self.install_test_root if post_install else self.stage.source_path,
             self.examples_src_dir)
         dims = self.spec.variants['dims'].value
-        inputs_nD = {'2': 'inputs_2d', '3': 'inputs_3d', 'rz': 'inputs_2d_rz'}
+        inputs_nD = {'1': 'inputs_1d', '2': 'inputs_2d', '3': 'inputs_3d', 'rz': 'inputs_2d_rz'}
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/warpx/package.py]:
```suggestion
        inputs_nD = {'1': 'inputs_1d', '2': 'inputs_2d', '3': 'inputs_3d',
                     'rz': 'inputs_2d_rz'}
```

File: var/spack/repos/builtin/packages/py-picmistandard/package.py [hunk 2]
```diff
@@ -10,16 +10,26 @@ class PyPicmistandard(PythonPackage):
     """Standard input format for Particle-In-Cell codes"""
 
     homepage = "https://picmi-standard.github.io"
-    url      = "https://github.com/picmi-standard/picmi/archive/refs/tags/0.0.14.tar.gz"
     git      = "https://github.com/picmi-standard/picmi.git"
+    pypi     = 'picmistandard/picmistandard-0.0.17.tar.gz'
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py]:
```suggestion
    pypi     = 'picmistandard/picmistandard-0.0.18.tar.gz'
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py]:
```suggestion
    pypi     = 'picmistandard/picmistandard-0.0.18.tar.gz'
```

File: var/spack/repos/builtin/packages/py-picmistandard/package.py [hunk 3]
```diff
@@ -10,16 +10,26 @@ class PyPicmistandard(PythonPackage):
...
     version('develop', branch='master')
-    version('0.0.14', sha256='b7eefdae1c43119984226b2df358c86fdeef7495084e47b3575e3d07e790ba30')
+    version('0.0.18', sha256='68c208c0c54b4786e133bb13eef0dd4824998da4906285987ddee84e6d195e71')
+    # 0.15 - 0.17 have broken install logic: missing requirements.txt on pypi
+    version('0.0.16', sha256='b7eefdae1c43119984226b2df358c86fdeef7495084e47b3575e3d07e790ba30',
+            url='https://github.com/picmi-standard/picmi/archive/refs/tags/0.0.14.tar.gz')
+    version('0.0.14', sha256='8f83b25b281fc0309a0c4f75c7605afd5fa0ef4df3b3ac115069478c119bc8c3')
 
     depends_on('python@3.6:', type=('build', 'run'))
     depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py]:
```suggestion
    depends_on('py-numpy@1.15:1', type=('build', 'run'))
    depends_on('py-scipy@1.5:1', type=('build', 'run'))
```
Haven't checked older versions, but this is the requirement in 0.0.18.

File: var/spack/repos/builtin/packages/py-warpx/package.py [hunk 4]
```diff
@@ -50,7 +50,8 @@ class PyWarpx(PythonPackage):
     depends_on('py-numpy@1.15.0:1', type=('build', 'run'))
     depends_on('py-mpi4py@2.0.0:', type=('build', 'run'), when='+mpi')
     depends_on('py-periodictable@1.5:1', type=('build', 'run'))
-    depends_on('py-picmistandard@0.0.14', type=('build', 'run'))
+    depends_on('py-picmistandard@0.0.14', type=('build', 'run'), when='@21.03:21.11')
+    depends_on('py-picmistandard@0.0.16', type=('build', 'run'), when='@21.12')
     depends_on('py-setuptools@38.6:', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py]:
```suggestion
    depends_on('py-setuptools@42:', type='build')
    depends_on('py-cmake@3.15:3', type='build')
```
From the `pyproject.toml`

File: var/spack/repos/builtin/packages/py-warpx/package.py [hunk 5]
```diff
@@ -50,7 +50,8 @@ class PyWarpx(PythonPackage):
     depends_on('py-numpy@1.15.0:1', type=('build', 'run'))
     depends_on('py-mpi4py@2.0.0:', type=('build', 'run'), when='+mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py]:
```suggestion
    depends_on('py-mpi4py@2.1.0:', type=('build', 'run'), when='+mpi')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-picmistandard/package.py [hunk 6]
```diff
@@ -10,16 +10,27 @@ class PyPicmistandard(PythonPackage):
     """Standard input format for Particle-In-Cell codes"""
 
     homepage = "https://picmi-standard.github.io"
-    url      = "https://github.com/picmi-standard/picmi/archive/refs/tags/0.0.14.tar.gz"
     git      = "https://github.com/picmi-standard/picmi.git"
+    pypi     = 'picmistandard/picmistandard-0.0.18.tar.gz'
+
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py:15]:
```suggestion
```

**Comment by ax3l** [comment]
Ready for merge :)

File: var/spack/repos/builtin/packages/py-warpx/package.py [hunk 7]
```diff
@@ -48,10 +48,12 @@ class PyWarpx(PythonPackage):
 
     depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py]:
```suggestion
    depends_on('python@3.6:3.9', type=('build', 'run'))
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**